### ISSUE TITLE
Restart Simulations: GPUs with/without Particles Hang

### ISSUE THREAD

**Issue by PrometheusPi** [ComputationalRadiationPhysics/picongpu#609]
### Update of the issue

Restarting simulations where some GPUs have particles and other have no particles are hanging.

**Verified with examples:**
- Bunch, no-particle-GPUs at "vacuum" GPUs around the bunch
- LWFA, no-particle-GPUs on the "first GPUs before the profile starts"

**Does the problem affect the `master` branch**?

- [x] ~~likely, must check~~ yep

---

### First found in the Bunch example

With the current `dev` the restart of the `examples/Bunch` does not work.

Running the simulation and dumping the restart files appear to work fine. However restarting it leads to a hanging simulation.
The picongpu simulation hangs right after printing:
```bash
...
[1,0]<stdout>:PIConGPUVerbose PHYSICS(1) | UNIT_EFIELD 2.6633e+13
[1,0]<stdout>:PIConGPUVerbose PHYSICS(1) | UNIT_BFIELD 88838
[1,0]<stdout>:PIConGPUVerbose PHYSICS(1) | UNIT_ENERGY 1.39727e-10
```
`[1,0]<stdout>:initialization time: ...` is never reached. 

In order to test this, I used the default `Bunch` example and added `--checkpoints 3000`.
I also removed all plugins and disabled radiation to strip the simulation of as much unneeded kernels as possible.

With `SPLASH_VERBOSE=100` enabled (thanks @f-schmitt-zih, that's an ubercool feature) the following output is produced last after restart:
```bash
...
[1,3]<stderr>:[SPLASH_LOG:3] DCDataSet::read (particles_info)
[1,3]<stderr>:[SPLASH_LOG:3] 
[1,3]<stderr>:
[1,3]<stderr>: ndims         = 1
[1,3]<stderr>: logical_size  = (32,1,1)
[1,3]<stderr>: physical_size = (32,1,1)
[1,3]<stderr>: dstBuffer     = (32,1,1)
[1,3]<stderr>: dstOffset     = (0,0,0)
[1,3]<stderr>: srcSize       = (32,1,1)
[1,3]<stderr>: srcOffset     = (0,0,0)
[1,3]<stderr>:
[1,0]<stderr>:[SPLASH_LOG:0]  returns ndims = 1
[1,0]<stderr>:[SPLASH_LOG:0]  returns sizeRead = (32,1,1)
[1,3]<stderr>:[SPLASH_LOG:3]  returns ndims = 1
[1,3]<stderr>:[SPLASH_LOG:3]  returns sizeRead = (32,1,1)
[1,2]<stderr>:[SPLASH_LOG:2]  returns ndims = 1
[1,2]<stderr>:[SPLASH_LOG:2]  returns sizeRead = (32,1,1)
[1,1]<stderr>:[SPLASH_LOG:1]  returns ndims = 1
[1,1]<stderr>:[SPLASH_LOG:1]  returns sizeRead = (32,1,1)
[1,0]<stderr>:[SPLASH_LOG:0] closing parallel data collector
[1,1]<stderr>:[SPLASH_LOG:1] closing parallel data collector
[1,2]<stderr>:[SPLASH_LOG:2] readDataSet
[1,3]<stderr>:[SPLASH_LOG:3] readDataSet
[1,2]<stderr>:[SPLASH_LOG:2] DCDataSet::read (x)
[1,2]<stderr>:[SPLASH_LOG:2] 
[1,2]<stderr>: ndims         = 1
[1,2]<stderr>: logical_size  = (14435572,1,1)
[1,2]<stderr>: physical_size = (14435572,1,1)
[1,2]<stderr>: dstBuffer     = (662223,1,1)
[1,2]<stderr>: dstOffset     = (0,0,0)
[1,2]<stderr>: srcSize       = (662223,1,1)
[1,2]<stderr>: srcOffset     = (0,0,0)
[1,2]<stderr>:
[1,3]<stderr>:[SPLASH_LOG:3] DCDataSet::read (x)
[1,3]<stderr>:[SPLASH_LOG:3] 
[1,3]<stderr>: ndims         = 1
[1,3]<stderr>: logical_size  = (14435572,1,1)
[1,3]<stderr>: physical_size = (14435572,1,1)
[1,3]<stderr>: dstBuffer     = (685264,1,1)
[1,3]<stderr>: dstOffset     = (0,0,0)
[1,3]<stderr>: srcSize       = (685264,1,1)
[1,3]<stderr>: srcOffset     = (662223,0,0)
```

**Comment by ax3l** [comment]
fixed in `dev` with #611

**Comment by PrometheusPi** [comment]
:+1:
